sentry/src/main/java/io/sentry/ServerNameResolvingEventProcessor.java

```diff
@@ -0,0 +1,134 @@
+package io.sentry;
+
+import io.sentry.util.Objects;
+import java.net.InetAddress;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * An {@link EventProcessor} that sets the local ip address to an {@link
+ * SentryEvent#getServerName()} property.
+ *
+ * <p>Note - this event processor is not used by SDK by default and must be configured manually on
+ * {@link SentryOptions} if there is intention to use it.
+ */
+public final class ServerNameResolvingEventProcessor implements EventProcessor {
+
+  /**
+   * Duration of the hostname caching.
+   *
+   * @see HostnameCache
+   */
+  private static final long HOSTNAME_CACHE_DURATION = TimeUnit.HOURS.toMillis(5);
```

Comment:
is it a common thing to change the server name? 5h seems too little, looks like also something we could have on the options, but not worth it at the moment

Reply:
I honestly don't know. This is how it was implemented in 1.x so I believe there was a reason for that.

Reply:
I see, makes sense, not sure if it's worth copying blindly tbh, it's a lot of code and complexity for a simple workaround that would be restarting/redeploying the App, of course, it's suboptimal, but a redeploy would happen soon or later right.

thoughts @bruno-garcia ?